File: typescript/tests/integration/region-id.integration.test.ts

import { AgentBay } from "../../src";
import { getTestApiKey } from "../utils/test-helpers";

describe("Region ID Integration Tests", () => {
  let apiKey: string;

  beforeAll(() => {
    apiKey = getTestApiKey();
  });

  describe("AgentBay initialization", () => {
    test("should create AgentBay client with region_id", () => {
      const config = {
        endpoint: "wuyingai.cn-shanghai.aliyuncs.com",
        timeout_ms: 60000,
        region_id: "cn-hangzhou"
      };
      const client = new AgentBay({
        apiKey,
        config: config
      });

      expect(client.getRegionId()).toBe("cn-hangzhou");
    });
    test("should handle empty region_id", () => {
      const config = {
        endpoint: "wuyingai.cn-shanghai.aliyuncs.com",
        timeout_ms: 60000,
        region_id: ""
      };
      const client = new AgentBay({
        apiKey,
        config: config
      });

      expect(client.getRegionId()).toBe("");
    });
  });

  describe("Session creation with region_id", () => {
    test("should create session successfully with region_id", async () => {
      const config = {
        endpoint: "wuyingai.cn-shanghai.aliyuncs.com",
        timeout_ms: 60000,
        region_id: "cn-hangzhou"
      };
      const client = new AgentBay({
        apiKey,
        config: config
      });

      const sessionResult = await client.create({imageId: "linux_latest"});

      expect(sessionResult.success).toBe(true);
      expect(sessionResult.session).toBeDefined();
      expect(sessionResult.session!.sessionId).toBeDefined();

      // Clean up
      if (sessionResult.session) {
        await sessionResult.session.delete();
      }
    }, 60000);

    test("should create session successfully without region_id", async () => {
      const client = new AgentBay({ apiKey });

      const sessionResult = await client.create({imageId: "linux_latest"});

      expect(sessionResult.success).toBe(true);
      expect(sessionResult.session).toBeDefined();
      expect(sessionResult.session!.sessionId).toBeDefined();

      // Clean up
      if (sessionResult.session) {
        await sessionResult.session.delete();
      }
    }, 60000);
  });

  describe("Context operations with region_id", () => {
    test("should create context successfully with region_id", async () => {
      const config = {
        endpoint: "wuyingai.cn-shanghai.aliyuncs.com",
        timeout_ms: 60000,
        region_id: "cn-hangzhou"
      };
      const client = new AgentBay({
        apiKey,
        config: config
      });

      const contextName = `test-context-typescript-region-id-${Date.now()}`;
      const contextResult = await client.context.get(contextName, true);

      expect(contextResult.success).toBe(true);
      expect(contextResult.context).toBeDefined();
      expect(contextResult.context!.id).toBeDefined();
      expect(contextResult.context!.name).toBe(contextName);

      // Clean up
      if (contextResult.context) {
        await client.context.delete(contextResult.context);
      }
    }, 30000);

    test("should get existing context without create (no LoginRegionId)", async () => {
      const config = {
        endpoint: "wuyingai.cn-shanghai.aliyuncs.com",
        timeout_ms: 60000,
        region_id: "cn-hangzhou"
      };
      const client = new AgentBay({
        apiKey,
        config: config
      });

      // Try to get non-existent context without create
      const contextName = `non-existent-context-${Date.now()}`;
      const contextResult = await client.context.get(contextName, false);

      // Should fail for non-existent context
      expect(contextResult.success).toBe(false);
      expect(contextResult.context).toBeUndefined();
    }, 30000);

    test("should create context successfully without region_id", async () => {
      const client = new AgentBay({ apiKey });

      const contextName = `test-context-no-region-${Date.now()}`;
      const contextResult = await client.context.get(contextName, true);

      expect(contextResult.success).toBe(true);
      expect(contextResult.context).toBeDefined();
      expect(contextResult.context!.id).toBeDefined();

      // Clean up
      if (contextResult.context) {
        await client.context.delete(contextResult.context);
      }
    }, 30000);
  });

  describe("End-to-end workflow with region_id", () => {
    test("should complete full workflow with region_id", async () => {
      const config = {
        endpoint: "wuyingai.cn-shanghai.aliyuncs.com",
        timeout_ms: 60000,
        region_id: "cn-hangzhou"
      };
      const client = new AgentBay({
        apiKey,
        config: config
      });

      // 1. Create context
      const contextName = `e2e-test-context-${Date.now()}`;
      const contextResult = await client.context.get(contextName, true);
      expect(contextResult.success).toBe(true);

      // 2. Create session
      const sessionResult = await client.create({imageId: "linux_latest"});
      expect(sessionResult.success).toBe(true);
      expect(sessionResult.session).toBeDefined();

      // 3. Get session info
      const sessionInfo = await sessionResult.session.info();
      const resultSessionInfo = sessionInfo.data
      expect(sessionInfo.success).toBe(true);
      expect(resultSessionInfo.sessionId).toBe(sessionResult.session!.sessionId);

      // 4. Clean up
      if (contextResult.context) {
        await client.context.delete(contextResult.context);
      }
      if (sessionResult.session) {
        await sessionResult.session.delete();
      }
    }, 90000);
  });
});

File: golang/tests/integration/region_id_integration_test.go
package integration_test

import (
	"os"
	"testing"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
)

// TestRegionIDIntegration tests region_id functionality end-to-end
func TestRegionIDIntegration(t *testing.T) {
	// Skip if no API key
	apiKey := os.Getenv("AGENTBAY_API_KEY")
	if apiKey == "" {
		t.Skip("AGENTBAY_API_KEY not set, skipping integration test")
	}

	t.Run("CreateAgentBayWithRegionID", func(t *testing.T) {
		// Test creating AgentBay client with region_id
		client, err := agentbay.NewAgentBay(apiKey, agentbay.WithRegionID("cn-hangzhou"))
		if err != nil {
			t.Fatalf("Failed to create AgentBay client: %v", err)
		}

		// Verify region_id is set
		if client.RegionID != "cn-hangzhou" {
			t.Errorf("Expected RegionID to be 'cn-hangzhou', got '%s'", client.RegionID)
		}
	})

	t.Run("CreateAgentBayWithoutRegionID", func(t *testing.T) {
		// Test creating AgentBay client without region_id
		client, err := agentbay.NewAgentBay(apiKey)
		if err != nil {
			t.Fatalf("Failed to create AgentBay client: %v", err)
		}

		// Verify region_id is empty
		if client.RegionID != "" {
			t.Errorf("Expected RegionID to be empty, got '%s'", client.RegionID)
		}
	})

	t.Run("CreateSessionWithRegionID", func(t *testing.T) {
		// Test session creation with region_id
		client, err := agentbay.NewAgentBay(apiKey, agentbay.WithRegionID("cn-beijing"))
		if err != nil {
			t.Fatalf("Failed to create AgentBay client: %v", err)
		}

		// Create session with default parameters
		params := agentbay.NewCreateSessionParams()
		result, err := client.Create(params)
		if err != nil {
			t.Fatalf("Failed to create session: %v", err)
		}

		if !result.Success {
			t.Fatalf("Session creation failed")
		}

		if result.Session == nil {
			t.Fatalf("Session is nil")
		}

		// Clean up
		defer func() {
			_, deleteErr := result.Session.Delete()
			if deleteErr != nil {
				t.Logf("Warning: Failed to delete session: %v", deleteErr)
			}
		}()

		t.Logf("✅ Successfully created session %s with region_id cn-beijing", result.Session.SessionID)
	})

	t.Run("CreateContextWithRegionID", func(t *testing.T) {
		// Test context creation with region_id
		client, err := agentbay.NewAgentBay(apiKey, agentbay.WithRegionID("cn-shenzhen"))
		if err != nil {
			t.Fatalf("Failed to create AgentBay client: %v", err)
		}

		// Create context
		contextName := "test-context-with-region-id"
		contextResult, err := client.Context.Get(contextName, true)
		if err != nil {
			t.Fatalf("Failed to create context: %v", err)
		}

		if !contextResult.Success {
			t.Fatalf("Context creation failed: %s", contextResult.ErrorMessage)
		}

		if contextResult.Context == nil {
			t.Fatalf("Context is nil")
		}

		// Clean up
		defer func() {
			_, deleteErr := client.Context.Delete(contextResult.Context)
			if deleteErr != nil {
				t.Logf("Warning: Failed to delete context: %v", deleteErr)
			}
		}()

		t.Logf("✅ Successfully created context %s with region_id cn-shenzhen", contextResult.Context.ID)
	})

	t.Run("GetContextWithoutCreate", func(t *testing.T) {
		// Test context get without create (should not pass LoginRegionId)
		client, err := agentbay.NewAgentBay(apiKey, agentbay.WithRegionID("cn-shenzhen"))
		if err != nil {
			t.Fatalf("Failed to create AgentBay client: %v", err)
		}

		// Try to get non-existent context without create
		contextName := "non-existent-context"
		contextResult, err := client.Context.Get(contextName, false)
		if err != nil {
			t.Fatalf("Unexpected error when getting non-existent context: %v", err)
		}

		// Should return success=false for non-existent context
		if contextResult.Success {
			t.Errorf("Expected context get to fail for non-existent context, but it succeeded")
		}

		t.Logf("✅ Context get without create behaved correctly for non-existent context")
	})
}